testing/cli-e2etest/testscenarios/test/list_test_test.go

```diff
@@ -110,102 +108,86 @@ func TestListTests(t *testing.T) {
-	//		require.Equal("The", oneMoreTestVars.Spec.Values[1].Key)
-	//		require.Equal("Third", oneMoreTestVars.Spec.Values[1].Value)
-	//	})
+	t.Run("list with JSON format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+
+		// When I try to list these tests by a valid field and in JSON format
+		// Then I should receive three tests
+		result := tracetestcli.Exec(t, "list test --sortBy name --sortDirection asc --output json", tracetestcli.WithCLIConfig(cliConfig))
+		helpers.RequireExitCodeEqual(t, result, 0)
+
+		testVarsList := helpers.UnmarshalJSON[types.ResourceList[types.TestResource]](t, result.StdOut)
+		require.Len(testVarsList.Items, 2)
+		require.Equal(len(testVarsList.Items), testVarsList.Count)
+
+		importTest := testVarsList.Items[0]
+		assert.Equal("Test", importTest.Type)
+		assert.Equal("RXrbV__4g", importTest.Spec.ID)
+		assert.Equal("Pokeshop - Import", importTest.Spec.Name)
+		assert.Equal("Import a Pokemon", importTest.Spec.Description)
+		assert.Equal("http", importTest.Spec.Trigger.Type)
+		assert.Equal("http://demo-api:8081/pokemon/import", importTest.Spec.Trigger.HTTPRequest.URL)
+		assert.Equal("POST", importTest.Spec.Trigger.HTTPRequest.Method)
+		assert.Equal(`{"id":52}`, importTest.Spec.Trigger.HTTPRequest.Body)
+		require.Len(importTest.Spec.Trigger.HTTPRequest.Headers, 1)
+		assert.Equal("Content-Type", importTest.Spec.Trigger.HTTPRequest.Headers[0].Key)
+		assert.Equal("application/json", importTest.Spec.Trigger.HTTPRequest.Headers[0].Value)
+
+		listTest := testVarsList.Items[1]
+		assert.Equal("Test", listTest.Type)
+		assert.Equal("fH_8AulVR", listTest.Spec.ID)
+		assert.Equal("Pokeshop - List", listTest.Spec.Name)
+		assert.Equal("List Pokemon", listTest.Spec.Description)
+		assert.Equal("http", listTest.Spec.Trigger.Type)
+		assert.Equal("http://demo-api:8081/pokemon?take=20&skip=0", listTest.Spec.Trigger.HTTPRequest.URL)
+		assert.Equal("GET", listTest.Spec.Trigger.HTTPRequest.Method)
+		assert.Equal("", listTest.Spec.Trigger.HTTPRequest.Body)
+		require.Len(listTest.Spec.Trigger.HTTPRequest.Headers, 1)
+		assert.Equal("Content-Type", listTest.Spec.Trigger.HTTPRequest.Headers[0].Key)
+		assert.Equal("application/json", listTest.Spec.Trigger.HTTPRequest.Headers[0].Value)
+	})
+
+	t.Run("list with pretty format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+
+		// When I try to list these tests by a valid field and in pretty format
+		// Then it should print a table with 6 lines printed: header, separator, three envs and empty line
+		result := tracetestcli.Exec(t, "list test --sortBy name --sortDirection asc --output pretty", tracetestcli.WithCLIConfig(cliConfig))
+		helpers.RequireExitCodeEqual(t, result, 0)
+
+		lines := strings.Split(result.StdOut, "\n")
+		require.Len(lines, 5)
+
+		require.Contains(lines[2], "RXrbV__4g")
+		require.Contains(lines[3], "fH_8AulVR")
```

Comment:
Is there a way to parse and test the pretty format using `helpers.UnmarshalTable` like we do on transactions: https://github.com/kubeshop/tracetest/blob/main/testing/cli-e2etest/testscenarios/transaction/list_transactions_test.go

Can we use it here?

Reply:
yea, good catch. updated!